File: exercises_00/01.py
"""
    Program that:
        Escriba un programa que calcule la suma de 1+2+3+...+n.
"""

def main () -> None:
    range_to_sum: int = 0

    try:
        range_to_sum = int(input("Ingrese el rango qué quiera sumar: "))

        if range_to_sum <= 0:
            raise ValueError("El valor ingresado debe ser mayor a 0")

    except ValueError as e:
        print(f"ERROR EL VALOR ES INVALIDO:\n{type(e)}\n{e}")
        return
    except Exception as e:
        print(f"ERROR:\n{type(e)}\n{e}")
        return

    result: int = calc_range_sum(range_to_sum)

    print(f"Resultado: \n\tSuma del Rango: {result}")

def calc_range_sum (range_length: int) -> int:
    sum: int = 0

    for i in range(range_length):
        sum += i + 1

    return sum

if __name__ == '__main__':
    main()

File: c_exercises_00/01.c
/*
 * =====================================================================================
 *
 *       Filename:  01.c
 *
 *    Description: Escriba un programa que calcule la suma de 1+2+3+...+n. 
 *
 *        Version:  1.0
 *        Created:  08/30/2023 05:30:57 PM
 *       Compiler:  gcc
 *
 * =====================================================================================
 */

#include <stdio.h>

int main (void) {
  int rangeToSum = 0;
  int result = 0;

  scanf("%d", &rangeToSum);

  result = ( rangeToSum * (rangeToSum + 1) ) / 2;

  printf("%d", result);

  return 0;
}
